rp2040-hal/src/i2c/controller.rs

```diff
@@ -284,53 +284,14 @@ impl<T: Deref<Target = Block>, PINS> eh1::I2c for I2C<T, PINS, Controller> {
         Write::write(self, addr, bytes)
     }
 
-    fn write_iter<B>(&mut self, address: u8, bytes: B) -> Result<(), Self::Error>
-    where
-        B: IntoIterator<Item = u8>,
-    {
-        let mut peekable = bytes.into_iter().peekable();
-        let addr: u16 = address.into();
-        Self::validate(addr, Some(peekable.peek().is_none()), None)?;
-        self.setup(addr);
-
-        while let Some(tx) = peekable.next() {
-            self.write_internal(&[tx], peekable.peek().is_none())?
-        }
-        Ok(())
-    }
-
-    fn write_read(&mut self, addr: u8, bytes: &[u8], buffer: &mut [u8]) -> Result<(), Error> {
-        WriteRead::write_read(self, addr, bytes, buffer)
-    }
-
-    fn write_iter_read<B>(
-        &mut self,
-        address: u8,
-        bytes: B,
-        buffer: &mut [u8],
-    ) -> Result<(), Self::Error>
-    where
-        B: IntoIterator<Item = u8>,
-    {
-        let mut peekable = bytes.into_iter().peekable();
-        let addr: u16 = address.into();
-        Self::validate(addr, Some(peekable.peek().is_none()), None)?;
-        self.setup(addr);
-
-        for tx in peekable {
-            self.write_internal(&[tx], false)?
-        }
-        self.read_internal(buffer, true, true)
-    }
-
```

Comment:
The main argument I've seen against them being "it is hard/suboptimal to implement with a dma".
I think those method do have value, even if the embedded-hal decided to drop them.

Can we make them inherent method instead of getting rid of them?

(the kind of nice thing that can be done with such an api can be seen [here](https://github.com/ithinuel/sh1107-rs/blob/main/sh1107/src/lib.rs#L165-L225)).

Reply:
That's a good idea.

Should there only be inherent methods for functions missing from E-H, or should there be an inherent impl for everything, so people who don't use drivers based on e-h don't need to bring the e-h traits into scope?